festim/exports/txt_export.py

```diff
@@ -46,22 +45,61 @@ def when_is_next_time(self, current_time):
                 return time
         return None
 
-    def write(self, current_time, steady):
+    def is_it_first_time_to_export(self, current_time, nb_iteration):
+        if self.times is None:
+            return nb_iteration == 0
+        else:
+            return np.isclose(self.times[0], current_time)
+
+    def write(self, current_time, nb_iteration, steady):
         # create a DG1 functionspace
         V_DG1 = f.FunctionSpace(self.function.function_space().mesh(), "DG", 1)
 
         solution = f.project(self.function, V_DG1)
         if self.is_it_time_to_export(current_time):
-            filename = "{}/{}_{}s.txt".format(self.folder, self.label, current_time)
             if steady:
                 filename = "{}/{}_steady.txt".format(self.folder, self.label)
+                header = "x,t=steady"
+            else:
+                filename = "{}/{}_transient.txt".format(self.folder, self.label)
```

Comment:
If we are now writing one file for all timesteps, then I don't think we need two different filenames.

Instead, we could have an attribute `.filename` defined in the `__init__` method:

```python
self.filename = f"{self.folder}/{self.label}.txt"
```

Reply:
Maybe, just use a variable `filename` in `write()`? Otherwise, `test_create_folder` in `test_txt_export` fails

Reply:
Well in theory it doesn't make to redefine it every time we write

Reply:
ok, is `test_create_folder` needed? I think, creation of passed folder is checked in `test_file_exists`

Reply:
Yes we need to keep these tests. Maybe we can adapt them?

Reply:
I think I understand the reason why `test_file_exists` fails, it's because as `filename` is created at the instantiation of the class, if you modify the `folder` attribute afterwards it doesn't modify `filename.`

The solution to this is to have it as a `property` of the class.

To add a `property` attribute, simply add:
```python
    def __init__(self, field, label, folder, times=None) -> None:
        super().__init__(field=field)
        if times:
            self.times = sorted(times)
        else:
            self.times = times
        self.label = label
        self.folder = folder
        self._first_time = True

    @property
    def filename(self):
        return f"{self.folder}/{self.label}.txt"
```